atlasdb-api/src/main/java/com/palantir/atlasdb/persist/api/ReusablePersister.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persist.api;
+
+import com.palantir.common.persist.Persistable.Hydrator;
+
+/**
+ * {@link ReusablePersister}s are required to have a no arg constructor and will be re-used across executions.
```

Comment:
Given the way we use it, we should probably also add that they must be thread-safe (I'm a bit worried "across executions" may make people think that these are sequential in some way)